### PR TITLE
Add rule check-for-sandbox-via-mac-address-ouis-in-dotnet

### LINKED ISSUE #465: add coverage of bad MAC addresses checked by SharpShooter framework
check performed [here](https://github.com/mdsecactivebreach/SharpShooter/blob/9aea0b7a723b61c01c21f1e4b55409edf39a5a3b/CSharpShooterStageless/CheckPlease.cs#L77).

we might already cover these but let's double-check.

### PR DESCRIPTION
**Pull request by jlmndt**
closes #465

### CODE REVIEW

File: nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - format: dotnet
+      - basic block:
+        - and:
+          - 3 or more:
+            - string: /00[:-]?0c[:-]?29/i
+            - string: /00[:-]?1c[:-]?14/i
+            - string: /00[:-]?50[:-]?56/i
+              description: VMware OUIs
+            - string: /08[:-]?00[:-]?27/i
+              description: PCS Systemtechnik GmbH OUI
```
Comment (mr-tz) [inline nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml:26]: are there specifics on why this is listed here? which sandbox uses this?

File: nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+rule:
+  meta:
+    name: check for sandbox via MAC address OUIs in .NET
+    namespace: anti-analysis/anti-vm/vm-detection
+    authors:
+      - jonathanlepore@google.com
+    description: detects sandbox detection via mac address organizationally unique identifiers (OUIs).  Based off publicly available CSharpShooter/CheckPlease.cs
```
Comment (mr-tz) [inline nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml:7]:
```suggestion
    description: detects sandbox detection via mac address organizationally unique identifiers (OUIs). Based off publicly available CSharpShooter/CheckPlease.cs
```

File: nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml (hunk 1 above)
Comment (jlmndt) [reply nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml:26]: these came straight out of CSharpShooter's CheckPlease.cs source code.  it looks like the pcs systemtechnik oui is related to virtualbox.

File: nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      - format: dotnet
+      - basic block:
+        - and:
+          - 3 or more:
+            - string: /00[:-]?0c[:-]?29/i
+            - string: /00[:-]?1c[:-]?14/i
+            - string: /00[:-]?50[:-]?56/i
+              description: VMware OUIs
+            - string: /08[:-]?00[:-]?27/i
+              description: PCS Systemtechnik GmbH OUI
```
Comment (mr-tz) [inline nursery/check-for-sandbox-via-mac-address-ouis-in-dotnet.yml:26]:
```suggestion
              description: PCS Systemtechnik GmbH OUI, potentially related to VirtualBox
```

**Review by mr-tz — APPROVED** [review]

**Comment by mr-tz** [comment]
Thank you!

**Pull request merged**